### PR TITLE
Upgrade dependencies to their latest stable release

### PR DESCRIPTION
**Pull request by gerolf-da**
The outdated dependencies where determined by running `bazel run @maven//:outdated`.

Not all dependencies are brought up to the latest version available on Maven Central, for the following reasons:

Nobody dares to touch DB:
  com.oracle.database.jdbc:ojdbc8 [19.14.0.0 -> 21.7.0.0]
  com.oracle.database.jdbc.debug:ojdbc8_g [19.14.0.0 -> 21.7.0.0]
  com.h2database:h2 [2.1.210 -> 2.1.214]
  org.flywaydb:flyway-core [8.4.1 -> 9.4.0]

License issues beyond version 2.6.x:
  com.typesafe.akka:akka-actor_2.13 [2.6.18 -> 2.7.0-M3]
  com.typesafe.akka:akka-actor-testkit-typed_2.13 [2.6.18 -> 2.7.0-M3]
  com.typesafe.akka:akka-actor-typed_2.13 [2.6.18 -> 2.7.0-M3]
  com.typesafe.akka:akka-http_2.13 [10.2.8 -> 10.4.0-M1]
  com.typesafe.akka:akka-http-spray-json_2.13 [10.2.8 -> 10.4.0-M1]
  com.typesafe.akka:akka-http-testkit_2.13 [10.2.8 -> 10.4.0-M1]
  com.typesafe.akka:akka-slf4j_2.13 [2.6.18 -> 2.7.0-M3]
  com.typesafe.akka:akka-stream_2.13 [2.6.18 -> 2.7.0-M3]
  com.typesafe.akka:akka-stream-testkit_2.13 [2.6.18 -> 2.7.0-M3]
  com.typesafe.akka:akka-testkit_2.13 [2.6.18 -> 2.7.0-M3]

Not a stable release:
  com.chuusai:shapeless_2.13 [2.3.3 -> 2.4.0-M1]
  io.circe:circe-core_2.13 [0.14.3 -> 0.15.0-M1]
  io.circe:circe-generic_2.13 [0.14.3 -> 0.15.0-M1]
  io.circe:circe-parser_2.13 [0.14.3 -> 0.15.0-M1]
  org.scalameta:munit_2.13 [0.7.29 -> 1.0.0-M6]
  org.scalactic:scalactic_2.13 [3.2.14 -> 3.3.0-SNAP3]
  org.scalatest:scalatest_2.13 [3.2.14 -> 3.3.0-SNAP3]
  org.scalatestplus:scalacheck-1-15_2.13 [3.2.11.0 -> 3.3.0.0-SNAP3]
  org.scalatestplus:selenium-3-141_2.13 [3.2.10.0 -> 3.3.0.0-SNAP3]
  org.scalatestplus:testng-6-7_2.13 [3.2.10.0 -> 3.3.0.0-SNAP3]
  org.tpolecat:doobie-core_2.13 [0.13.4 -> 1.0.0-RC2]
  org.tpolecat:doobie-hikari_2.13 [0.13.4 -> 1.0.0-RC2]
  org.tpolecat:doobie-postgres_2.13 [0.13.4 -> 1.0.0-RC2]
  com.fasterxml.jackson.core:jackson-core [2.13.4 -> 2.14.0-rc2]
  com.fasterxml.jackson.core:jackson-databind [2.13.4 -> 2.14.0-rc2]

Netty + boringssl + grpc + protobuf need be kept in sync, and these are
also not stable releases:
  io.netty:netty-buffer [4.1.79.Final -> 5.0.0.Alpha2]
  io.netty:netty-codec-http2 [4.1.79.Final -> 5.0.0.Alpha2]
  io.netty:netty-handler [4.1.79.Final -> 5.0.0.Alpha2]
  io.netty:netty-handler-proxy [4.1.79.Final -> 5.0.0.Alpha2]
  io.netty:netty-resolver [4.1.79.Final -> 5.0.0.Alpha2]
  com.google.protobuf:protobuf-java [3.21.7 -> 4.0.0-rc-2]

Scalaz changed a few annoying things in 7.3, so let's stick to 7.2:
  org.scalaz:scalaz-core_2.13 [7.2.34 -> 7.4.0-M12]
  org.scalaz:scalaz-scalacheck-binding_2.13 [7.2.34-scalacheck-1.15 -> 7.4.0-M12]
  org.scalacheck:scalacheck_2.13 [1.15.4 -> 1.17.0]

Determined by the rules_scala:
  org.scala-lang:scala-library [2.13.8 -> 2.13.10]

CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

**Comment by gerolf-da** [comment]
@mziolekda, @miklos-da, @rgugliel-da, @soren-da: heads up that these dependency updates might be coming your way soon.

**Comment by rgugliel-da** [comment]
> @mziolekda, @miklos-da, @rgugliel-da, @soren-da: heads up that these dependency updates might be coming your way soon. If you have any concerns, please raise them rather sooner than later.

Thanks! In such cases, you can drop a message in channel #team-sdk-updates where we discuss bumping Daml SDK in Canton.

File: bazel-java-deps.bzl [hunk 1]
```diff
@@ -45,50 +45,51 @@ version_specific = {
...
+gatling_version = "3.8.4"
+guava_version = "31.1-jre"
 
 # observability libs
-dropwizard_version = "4.1.2"
-opentelemetry_version = "1.12.0"
-prometheus_version = "0.8.1"
+dropwizard_version = "4.2.12"
+opentelemetry_version = "1.19.0"
+prometheus_version = "0.16.0"
```
Comment (nicu-da) [inline bazel-java-deps.bzl:60]: @gerolf-da We would prefer to not update the observability libraries yet. 
Adding label support for metrics currently depends on the 1.12.0 update and we're in the process of selecting an update path. But the current version allows us to easily add label support alongside dropwizard. 
Also the dropwizard library doesn't work with the current setup without some changes to our init structure. 
Already available PR to upgrade telemetry to valid versions: https://github.com/digital-asset/daml/pull/15211
Issue

File: ledger/participant-integration-api/src/test/lib/scala/platform/indexer/ha/IndexerStabilityTestFixture.scala [hunk 2]
```diff
@@ -94,6 +93,7 @@ object IndexerStabilityTestFixture {
                       }
                     )
                     .acquire()
+                  metrics = new Metrics(new MetricRegistry)
```
Comment (nicu-da) [inline ledger/participant-integration-api/src/test/lib/scala/platform/indexer/ha/IndexerStabilityTestFixture.scala:96]: Why is this change required?

Comment (gerolf-da) [reply ledger/participant-integration-api/src/test/lib/scala/platform/indexer/ha/IndexerStabilityTestFixture.scala:96]: I'm actually surprised it wasn't required before actually. Using the same `MetricRegistry` instance for multiple indexers means that multiple indexers try to register a gauge under the same name (the crash/exception was triggered by InMemoryStateUpdater.scala#97). I did a quick search of what might have changed in Dropwizard Metrics between the two releases, but I would expect it to fail already actually.

File: bazel-java-deps.bzl (hunk 1 above)
Comment (gerolf-da) [reply bazel-java-deps.bzl:60]: Okay, I will revert these libraries then.

File: ledger/participant-integration-api/src/test/lib/scala/platform/indexer/ha/IndexerStabilityTestFixture.scala (hunk 2 above)
Comment (nicu-da) [reply ledger/participant-integration-api/src/test/lib/scala/platform/indexer/ha/IndexerStabilityTestFixture.scala:99]: Surprised why does it even pass though. 
I've created this issue for the newly added gauges in dropwizard https://github.com/dropwizard/metrics/issues/2920, but shouldn't this fail as well during indexer restarts? We're recreating the `DbDispatcher` as part of the restart therefore it would re-create the thread pool associated with it. And the `gauge` registration was added as part of metric instrumentation for the `InstrumentedExecutorService` for the `ThreadPoolExecutor` in this case.

**Review by tudor-da — APPROVED** [review]

File: ledger/participant-integration-api/src/test/lib/scala/platform/indexer/ha/IndexerStabilityTestFixture.scala (hunk 2 above)
Comment (atriantafyllos-da) [reply ledger/participant-integration-api/src/test/lib/scala/platform/indexer/ha/IndexerStabilityTestFixture.scala:99]: This is no longer relevant since InstrumentedExecutorService does not use dropwizard anymore.
